### PR TITLE
Update to candidate-java-sample from David Valen

### PR DESCRIPTION
**Pull request by dvalen14**
Here's my updates to the sample java file.   I have the sample passwords in a spreadsheet I can share via email or other method.  Email: david.valen@live.com

Thank you for the opportunity.  Hope to hear from you soon.

### CODE REVIEW

File: src/main/java/com/bravo/user/service/UserService.java [hunk 1]
```diff
@@ -49,7 +58,22 @@ public UserReadDto retrieve(final String id){
     LOGGER.info("found user '{}'", id);
     return resourceMapper.convertUser(user);
   }
-
+  
+  public PasswordValidateDto validate(final PasswordValidateDto passwordValidateDto, final PageRequest pageRequest, final HttpServletResponse httpResponse){
+      final String encryptedPassword = encryptService.encrypt(passwordValidateDto.getPassword());
+      User user = userRepository.findByEmail(passwordValidateDto.getEmail());
+      
+      if(!passwordValidateDto.getEmail().matches("^([_a-zA-Z0-9-]+(\\.[_a-zA-Z0-9-]+)*@[a-zA-Z0-9-]+(\\.[a-zA-Z0-9-]+)*(\\.[a-zA-Z]{1,6}))?$")){
```
Comment (tlynema-bravolt) [inline src/main/java/com/bravo/user/service/UserService.java:66]: Could this use the isEmailValid below?

File: src/main/java/com/bravo/user/utility/ValidatorUtil.java [hunk 2]
```diff
@@ -46,6 +46,14 @@ else if(value instanceof String){
     }
     return isValid;
   }
+  
+  public static boolean isEmailValid(final String string){
+    return string != null && string.matches("^([_a-zA-Z0-9-]+(\\.[_a-zA-Z0-9-]+)*@[a-zA-Z0-9-]+(\\.[a-zA-Z0-9-]+)*(\\.[a-zA-Z]{1,6}))?$");
```
Comment (tlynema-bravolt) [inline src/main/java/com/bravo/user/utility/ValidatorUtil.java:51]: string is a hard to follow name here.  The method says to pass in an email so it's followable, inside of the method, we can't tell that string is and email.

File: src/main/java/com/bravo/user/service/UserService.java [hunk 3]
```diff
@@ -49,7 +58,22 @@ public UserReadDto retrieve(final String id){
...
+      	throw new UserNamePasswordException("Invalid Email Format");
+      }
+      
+      if((user == null) || (!encryptedPassword.equals(user.getPassword()))){
```
Comment (tlynema-bravolt) [inline src/main/java/com/bravo/user/service/UserService.java:70]: Can this be done in a way that avoids a possible null return?

File: src/main/java/com/bravo/user/exception/UserNamePasswordException.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.bravo.user.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.UNAUTHORIZED)
+public class UserNamePasswordException extends RuntimeException {
+
+  public UserNamePasswordException(final String message){
+    super(String.format("UserName/Passworod Invalid | %s", message));
```
Comment (DaveAldon) [inline src/main/java/com/bravo/user/exception/UserNamePasswordException.java:10]: Simple misspelling here 😄

**Comment by DaveAldon** [comment]
Would like to mention, we're not expecting any further commits in response to our comments. But you're invited to elaborate/comment on any of our questions. Thank you for your PR!

File: src/main/java/com/bravo/user/service/UserService.java (hunk 1 above)
Comment (dvalen14) [reply src/main/java/com/bravo/user/service/UserService.java:66]: Agree...this should have used the isEmailValid from ValidatorUtil

if(ValidatorUtil.isEmailInvalid(passwordValidateDto.getEmail())){...

File: src/main/java/com/bravo/user/utility/ValidatorUtil.java (hunk 2 above)
Comment (dvalen14) [reply src/main/java/com/bravo/user/utility/ValidatorUtil.java:51]: Again...agree here.   The variable should be more descriptive probably using "email" for the string variable name.

File: src/main/java/com/bravo/user/exception/UserNamePasswordException.java (hunk 4 above)
Comment (dvalen14) [reply src/main/java/com/bravo/user/exception/UserNamePasswordException.java:10]: Agree...would update...

File: src/main/java/com/bravo/user/service/UserService.java (hunk 3 above)
Comment (dvalen14) [reply src/main/java/com/bravo/user/service/UserService.java:70]: Yes...using the included Lombok library, I should have done something like this:
      if(!(isPasswordValid(user, encryptedPassword)){
      	throw new UserNamePasswordException("Invalid Email/Password combination");
      }  
 
 private boolean isPasswordValid(@NonNull User user, @NonNull String encryptedPassword){
    return encryptedPassword.equals(user.getPassword());
  }

**Comment by dvalen14** [comment]
Thanks for the feedback.   I'm always learning.  At the very least, I'll use this in future projects.
